### PR TITLE
add Dockerfile for testing and deployment

### PR DESCRIPTION
**Pull request by chestercheng**
As title

The docker image include [` runtools `](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/runtools) extension, we can run all the cells in a notebook and skip over any exceptions.

And if anyone try to play with this project on localhost, they can run the following commands to setup the environment.

```
$ docker build --no-cache -t first_day_with_python .
$ docker run -it --name first_day_with_python -p 8888:8888 first_day_with_python
[I 13:50:22.387 NotebookApp] Writing notebook server cookie secret to /home/pyusr/.local/share/jupyter/runtime/notebook_cookie_secret
[I 13:50:22.657 NotebookApp] [jupyter_nbextensions_configurator] enabled 0.4.0
[I 13:50:22.658 NotebookApp] Serving notebooks from local directory: /home/pyusr
[I 13:50:22.658 NotebookApp] The Jupyter Notebook is running at:
[I 13:50:22.659 NotebookApp] http://(d782eaf6181a or 127.0.0.1):8888/?token=684796fcaa9594160bfc56ae3bdb0314c9c40c316df07dc6
[I 13:50:22.659 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 13:50:22.660 NotebookApp]

    Copy/paste this URL into your browser when you connect for the first time,
    to login with a token:
        http://(d782eaf6181a or 127.0.0.1):8888/?token=684796fcaa9594160bfc56ae3bdb0314c9c40c316df07dc6
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    NB_GRP=pyhug
+RUN groupadd -r ${NB_GRP}
+RUN useradd --no-log-init --create-home -r -g ${NB_GRP} ${NB_USR}
+
+# Running jupyter notebook as non-root user
+USER ${NB_USR}
+EXPOSE 8888
+WORKDIR /home/${NB_USR}
+ADD * ./
+CMD ["jupyter", "notebook", "--port=8888", "--no-browser", "--ip=0.0.0.0"]
```
Comment (yungyuc) [inline Dockerfile]: `0.0.0.0` is probably needed for a hosted service, but we should confirm it with infuseai.

**Comment by yungyuc** [comment]
Looks good.

**Comment by hlb** [comment]
@chestercheng we are using JupyterLab, so it would be great and instantly compatible if you start from jupyter/base-notebook https://hub.docker.com/r/jupyter/base-notebook/ and add notebook extension you need.

**Comment by chestercheng** [comment]
@hlb I've changed the base docker image from "python" to "jupyter/base-notebook" https://github.com/chestercheng/first_day_with_python/commit/a2e00a8021458d8ad74c061423c941adbef7c9c3 .
Please review it. Is it correct?

**Comment by yungyuc** [comment]
@chestercheng Why not file a PR?

**Comment by chestercheng** [comment]
Sorry, I forgot it... I will do that later.

**Pull request merged**